Towny/src/main/java/com/palmergames/bukkit/towny/utils/SpawnUtil.java

```diff
@@ -669,12 +674,52 @@ private static void initiateSpawn(Player player, Location spawnLoc, int cooldown
 			// Don't use teleport warmup
 			if (player.getVehicle() != null)
 				player.getVehicle().eject();
-			PaperLib.teleportAsync(player, spawnLoc, TeleportCause.COMMAND);
+			
+			List<Entity> pets = getPets(player);
+			pets.add(player);
+			teleportEntities(spawnLoc, pets, TeleportCause.COMMAND);
 			if (cooldown > 0 && !hasPerm(player, PermissionNodes.TOWNY_SPAWN_ADMIN_NOCOOLDOWN))
 				CooldownTimerTask.addCooldownTimer(player.getName(), "teleport", cooldown);
 		}
 	}
 
+	/**
+	 * Gets any pets owned by the specified player that are nearby them
+	 * 
+	 * @param player The player to get the pets of
+	 */
+	public static List<Entity> getPets(Player player) {
+		List<Entity> pets = new ArrayList<>();
+		
+		for (Entity entity : player.getNearbyEntities(16, 16, 16)) {
+			if (!(entity instanceof Tameable tameable)) continue;
+

```

Comment:
```suggestion
		for (Tameable tameable : player.getLocation().getNearbyEntitiesByType(Tameable.class, 16)) {
```

Reply:
I was looking at the paper javadocs, but the same method exists on World on spigot